frontend/app/src/components/molecules/nav-bar/banner.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Button } from '@/components/ui/button';
+
+export interface BannerProps {
+  message: React.ReactNode;
+  type?: 'info' | 'warning' | 'success' | 'error';
+  actionText?: string;
+  onAction?: () => void;
+  showBanner?: boolean;
+}
+
+export const Banner: React.FC<BannerProps> = ({
```

Comment:
stylistic nit: I've been defining components like this:

```typescript
export const Banner = ({foo, bar}: BannerProps) => {}
```

I don't care which we end up going for, but I think we should try to be consistent. WDYT? I kinda like the way I've been doing it because it's a little more terse, but I'm fine either way

Reply:
personally prefer React.FC<BannerProps> since it adds some additional type safety if needed later

Reply:
ah nice, vs. just `JSX.Element`? sounds good to me